src/middlewared/middlewared/plugins/chart_releases_linux/host_path_attachments.py

```diff
@@ -1,6 +1,73 @@
+import os
+
 from middlewared.common.attachment import FSAttachmentDelegate
+from middlewared.service import private, Service
 from middlewared.utils.path import is_child
 
+from .utils import is_ix_volume_path, safe_to_ignore_path
+
+
+class ChartReleaseService(Service):
+
+    class Config:
+        namespace = 'chart.release'
+
+    @private
+    async def ix_mount_validation(self, path, path_type):
+        path_list = [p for p in path.split('/') if p.strip()]
+        if safe_to_ignore_path(path):
+            if path_list and len(path_list) < 3 and path_list == 'mnt':
+                return f'Invalid path {path}. Mounting root dataset or path outside a pool is not allowed'
+        elif path_list and path_list[0] == 'cluster':
```

Comment:
Why are we using the fuse mount via host path instead of something like glusterfs CSI?

There is no protection for ctdb_shared_vol (which is an internal volume that should _never under any circumstances_ be accessed by any app).

Reply:
For that matter, we probably want to discourage users from exposing their entire glusterfs volume via host path.

`/cluster/glvol/` is probably a bad idea because it removes the ability of admin to be able to pick / choose which data in volume to expose to kubernetes. `/cluster/gvol/k8s` is a safer bet.

Reply:
There are some plans to move contents of ctdb_shared_vol to special dir inside root of first created gluster volume. This means actually we _must_ prevent exposing `/cluster/glvol` to apps.

Reply:
> Why are we using the fuse mount via host path instead of something like glusterfs CSI?
> 
> There is no protection for ctdb_shared_vol (which is an internal volume that should _never under any circumstances_ be accessed by any app).

I agree here, there are a number of ways to mount a gluster cluster directly using a CSI and those also integrate better in potential future usecases for actually clusterised kubernetes.